pystac/extensions/raster.py

```diff
@@ -0,0 +1,715 @@
+        if v is not None:
+            self.properties["histogram"] = v.to_dict()
+        else:
+            self.properties.pop("histogram", None)
+
+    def __repr__(self) -> str:
+        return "<Raster Band>"
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Returns the dictionary representing the JSON of this Band.
+
+        Returns:
+            dict: The wrapped dict of the Band that can be written out as JSON.
+        """
+        return self.properties
+
+
+class RasterExtension(
+    Generic[T], PropertiesExtension, ExtensionManagementMixin[pystac.Item]
+):
+    """RasterItemExt is the extension of the Item in the raster extension
+
+    Args:
+        item : The item to be extended.
+
+    Attributes:
+        item : The Item that is being extended.
+
+    Note:
+        Using RasterItemExt to directly wrap an item will add
+        the 'raster' extension ID to the item's stac_extensions.
+    """
+
+    def apply(self, bands: List[RasterBand]) -> None:
+        """Applies label extension properties to the extended Item.
+
+        Args:
+            bands : a list of :class:`~pystac.RasterBand` objects that represent
+                the available raster bands.
+        """
+        self.bands = bands
+
+    @property
+    def bands(self) -> Optional[List[RasterBand]]:
+        """Gets or sets a list of available bands where each item is a :class:`~RasterBand`
+        object (or ``None`` if no bands have been set). If not available the field
+        should not be provided.
+        """
+        return self._get_bands()
+
+    @bands.setter
+    def bands(self, v: Optional[List[RasterBand]]) -> None:
+        self._set_property(
+            BANDS_PROP, map_opt(lambda bands: [b.to_dict() for b in bands], v)
+        )
+
+    def _get_bands(self) -> Optional[List[RasterBand]]:
+        return map_opt(
+            lambda bands: [RasterBand(b) for b in bands],
+            self._get_property(BANDS_PROP, List[Dict[str, Any]]),
+        )
+
+    @classmethod
+    def get_schema_uri(cls) -> str:
+        return SCHEMA_URI
+
+    @staticmethod
+    def ext(obj: T) -> "RasterExtension[T]":
+        if isinstance(obj, pystac.Asset):
+            return cast(RasterExtension[T], AssetRasterExtension(obj))
+        else:
+            raise pystac.ExtensionTypeError(
+                f"Raster extension does not apply to type {type(obj)}"
+            )
+
+    @staticmethod
+    def summaries(obj: pystac.Collection) -> "SummariesRasterExtension":
+        return SummariesRasterExtension(obj)
+
+
+class AssetRasterExtension(RasterExtension[pystac.Asset]):
+    def __init__(self, asset: pystac.Asset):
+        self.asset_href = asset.href
+        self.properties = asset.properties
+        if asset.owner and isinstance(asset.owner, pystac.Item):
+            self.additional_read_properties = [asset.owner.properties]
+
+    def __repr__(self) -> str:
+        return "<AssetRasterExtension Item id={}>".format(self.asset_href)
```

Comment:
This was an error in the original label extension code...

```suggestion
        return "<AssetRasterExtension Asset href={}>".format(self.asset_href)
```